packages/cardano-services/src/Reward/RewardHttpService.ts

```diff
@@ -0,0 +1,61 @@
+import * as OpenApiValidator from 'express-openapi-validator';
+import { Cardano, EpochRewards, ProviderError, ProviderFailure, RewardHistoryProps } from '@cardano-sdk/core';
+import { DbSyncRewardProvider } from './DbSyncRewardProvider/DbSyncReward';
+import { HttpServer, HttpService } from '../Http';
+import { Logger, dummyLogger } from 'ts-log';
+import { ServiceNames } from '../Program';
+import { providerHandler } from '../util';
+import express from 'express';
+import path from 'path';
+
+export interface RewardServiceDependencies {
+  logger?: Logger;
+  rewardProvider: DbSyncRewardProvider;
+}
+
+export class RewardHttpService extends HttpService {
+  private constructor({ logger = dummyLogger }: RewardServiceDependencies, router: express.Router) {
+    super(ServiceNames.Reward, router, logger);
+  }
+
+  async healthCheck() {
+    return Promise.resolve({ ok: true });
+  }
+
+  static create({ logger = dummyLogger, rewardProvider }: RewardServiceDependencies) {
+    const router = express.Router();
+    const apiSpec = path.join(__dirname, 'openApi.json');
+    router.use(
+      OpenApiValidator.middleware({
+        apiSpec,
+        ignoreUndocumented: true,
+        validateRequests: true,
+        validateResponses: true
+      })
+    );
+    // Add initial healthCheck of the provider when implemented
+    router.post(
+      '/account-balance',
+      providerHandler<[Cardano.RewardAccount], bigint>(async ([rewardAccount], _, res) => {
```

Comment:
Catch block here is exactly the same for both handlers (and I suspect most other provider http services too). Maybe we could refactor it to hoist try-catch code to `providerHandler`, adding an optional argument like `errorHandler?: (error: unknown) => void`

Then maybe we could simplify it to something like this, as the success status code is always the same for all providers:
```suggestion
      providerHandler<[Cardano.RewardAccount], bigint>(async ([rewardAccount], _, res) => rewardProvider.rewardAccountBalance(rewardAccount))
```

Reply:
> Catch block here is exactly the same for both handlers (and I suspect most other provider http services too). Maybe we could refactor it to hoist try-catch code to `providerHandler`, adding an optional argument like `errorHandler?: (error: unknown) => void`
> 
> Then maybe we could simplify it to something like this, as the success status code is always the same for all providers:
> 
> This is actually duplicate with what @lgobbi-atix did [here](https://github.com/input-output-hk/cardano-js-sdk/pull/248#discussion_r877859900) so it can be resolved when merging

Sorry but this link redirects me to my current page. So, this suggestion may be already resolved in Leo's PR?

Reply:
Sorry must have pasted in the wrong link. [This](https://github.com/input-output-hk/cardano-js-sdk/pull/243/files#diff-d403c7615a0c015170e3bc860f71607b78c5a840ca10bfc832991ba1760f7d65R34) is the correct one. Yes, it resolves this suggestion